src/tensor_ops/upscale2d/mod.rs

```diff
@@ -74,53 +81,83 @@ pub trait Upscale2DWithMethod<M: UpscaleMethod>: HasErr {
     type Output<OH: Dim, OW: Dim>;
     fn try_upscale2d<const OH: usize, const OW: usize>(
         self,
+        method: M,
     ) -> Result<Self::Output<Const<OH>, Const<OW>>, Self::Err> {
-        self.try_upscale2d_like(Const::<OH>, Const::<OW>)
+        self.try_upscale2d_like(method, Const::<OH>, Const::<OW>)
     }
 
     fn try_upscale2d_like<OH: Dim, OW: Dim>(
         self,
+        method: M,
         height: OH,
         width: OW,
     ) -> Result<Self::Output<OH, OW>, Self::Err>;
 }
 
+/// Upscales an image to a new shape. Valid methods of upscaling are:
+///
+/// - [NearestNeighbor] pytorch equivalent: `F.interpolate(..., mode="nearest")`
+/// - [Bilinear] pytorch equivalent: `F.interpolate(..., mode="bilinear", align_corners=True)`
+///
+/// Compile time upscale:
+/// ```rust
+/// # use dfdx::prelude::*;
+/// # let dev: Cpu = Default::default();
+/// let t: Tensor<Rank3<3, 32, 32>, f32, _> = dev.zeros();
+/// let y: Tensor<Rank3<3, 64, 64>, f32, _> = t.upscale2d(NearestNeighbor);
+/// ```
+///
+/// Runtime upscale:
+/// ```rust
+/// # use dfdx::prelude::*;
+/// # let dev: Cpu = Default::default();
+/// let t: Tensor<Rank3<3, 32, 32>, f32, _> = dev.zeros();
+/// let y: Tensor<(Const<3>, usize, usize), f32, _> = t.upscale2d_like(NearestNeighbor, 64, 64);
```

Comment:
The interface to upscale2d has changed. It now takes in the method as a parameter so you can call it like in these examples (notably without specifying any generics in either case). This unifies how you call them in both const/runtime situations, as before you would've had to specify the method in the generic both ways